plugins/user/tjreportsindexer/tjreportsindexer.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/**
+ * @package     TJReports
+ * @subpackage  com_tjreports.tjreportsfields
+ *
+ * @copyright   Copyright (C) 2009 - 2019 Techjoomla. All rights reserved.
+ * @license     http://www.gnu.org/licenses/gpl-2.0.html GNU/GPL
+ */
+
+// No direct access
+defined('_JEXEC') or die('Restricted access');
+
+/**
+ * Class for Tjreportsindexer User Plugin
+ *
+ * @since  1.1.0
+ */
+class PlgUserTjreportsindexer extends JPlugin
+{
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 *
+	 * @since  3.2.11
+	 */
+	protected $autoloadLanguage  = true;
+
+	protected $customFieldsTable = '#__tjreports_com_users_user';
+
+	protected $listTypeFields     = array ('checkboxes', 'list', 'radio');
+
+	protected $listTypeCoreFields = array ('user', 'usergrouplist');
+
+	/**
+	 * Utility method to act on a user after it has been saved.
+	 *
+	 * This method sends a registration email to new users created in the backend.
+	 *
+	 * @param   array    $user     Holds the new user data.
+	 * @param   boolean  $isNew    True if a new user is stored.
+	 * @param   boolean  $success  True if user was successfully stored in the database.
+	 * @param   string   $msg      Message.
+	 *
+	 * @return  void
+	 *
+	 * @since   1.6
+	 */
+	public function onUserAfterSave($user, $isNew, $success, $msg)
+	{
+		/*
+		  [com_fields] => Array
+		(
+			[job-position] => Developer
+		)*/
+
+		if (empty($user['com_fields']))
+		{
+			return;
+		}
+
+		// Delete existing user-data entry
+		// Here record_id = user_id
+		$this->deleteIndexerEntry($user['id']);
+
+		// Add new entry
+		$this->addIndexerEntry($user);
+	}
+
+	/**
+	 * Remove all sessions for the user name
+	 *
+	 * Method is called after user data is deleted from the database
+	 *
+	 * @param   array    $user     Holds the user data
+	 * @param   boolean  $success  True if user was successfully stored in the database
+	 * @param   string   $msg      Message
+	 *
+	 * @return  boolean
+	 *
+	 * @since   1.6
+	 */
+	public function onUserAfterDelete($user, $success, $msg)
+	{
+		if (!$success)
+		{
+			return false;
+		}
+
+		$this->deleteIndexerEntry($user['id']);
+	}
+
+	/**
+	 * Add Indexer Entry
+	 *
+	 * @param   array  $user  Holds the user data
+	 *
+	 * @return  void
+	 *
+	 * @since  1.1.0
+	 */
+	protected function addIndexerEntry($user)
+	{
+		$db = JFactory::getDbo();
+
+		// Get column name, type for custom fields index table
+		$columnsDetails = $db->getTableColumns($this->customFieldsTable);
+
+		// Extract column names from $columnDetails
+		$columnNames = array();
+
+		foreach ($columnsDetails as $key => $val)
+		{
+			$columnNames[] = $key;
+		}
+
+		// Get field names
+		$fieldsPosted = array_keys($user['com_fields']);
+
+		// For all fields get type, fieldparams
+		$query = $db->getQuery(true);
+
+		// Prepare the insert query.
+		$query
```

Comment:
Use joomla code to fetch field values. You can then filter out the ones needed.

Reply:
yup

Reply:
yup